grid-proxy/internal/indexer/indexer.go

```diff
@@ -0,0 +1,54 @@
+package indexer
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/tfgrid-sdk-go/rmb-sdk-go/peer"
+)
+
+const (
+	indexerCallTimeout = 10 * time.Second
+)
+
+type Watcher interface {
+	Start(ctx context.Context)
+}
+
+type Indexer struct {
+	Watchers  map[string]Watcher
+	Paused    bool
+	Context   context.Context
+	RmbClient *peer.RpcClient
+}
+
+func NewIndexer(
+	ctx context.Context,
+	paused bool,
+	rmbClient *peer.RpcClient,
+) *Indexer {
+	return &Indexer{
+		Watchers:  make(map[string]Watcher),
+		Paused:    paused,
+		Context:   ctx,
+		RmbClient: rmbClient,
+	}
+}
+
+func (i *Indexer) RegisterWatcher(name string, watcher Watcher) {
+	i.Watchers[name] = watcher
+}
+
+func (i *Indexer) Start() {
+	if i.Paused {
```

Comment:
It makes sense to resume after pausing, not sure why you return

Reply:
actually `pause` was a misleading i need a flag to stop the indexer in this run, it is helpful when running integration tests to keep data unchanged from the indexer.

for a better naming i called the flag `-no-indexer` also made the decision for starting manager or not outside the pkg in the main file